src/server/frontend_xwayland/xwayland_wm.cpp

```diff
@@ -362,6 +362,8 @@ void mf::XWaylandWM::handle_events()
 {
     bool got_events = false;
 
+    connection->verify_not_in_error_state();
```

Comment:
Not blocking, but I'd generally find it more obvious to just not *call* `handle_events` when the connection has errored.